pkg/agent/dummy_data_gatherer.go

```diff
@@ -22,9 +22,18 @@ func (c *dummyConfig) NewDataGatherer(ctx context.Context) (datagatherer.DataGat
 }
 
 type dummyDataGatherer struct {
-	Param1 string
+	Param1        string
+	attemptNumber int
 }
 
 func (c *dummyDataGatherer) Fetch() (interface{}, error) {
-	return nil, nil
+	var err error
+	if c.attemptNumber < 2 {
```

Comment:
This should be configurable, not hardcoded.

I suggest that you add a config parameter to `dummyConfig` that let's us configure the behaviour from the configuration file.

Reply:
That's a good point! I hadn't thought about the most general use case there. I can add that to the data gatherers parameters. i.e. `failedAttempts int`